tools/mc-java-protoc/src/main/java/io/spine/tools/mc/java/protoc/field/FieldGen.java

```diff
@@ -73,27 +71,61 @@ private FieldGen(ImmutableList<CodeGenerationTask> tasks) {
     /**
      * Creates a new instance based on the passed Protoc config.
      */
-    @SuppressWarnings("MethodWithMultipleLoops") // Required to configure code generation tasks.
     public static FieldGen instance(SpineProtocConfig spineProtocConfig) {
         checkNotNull(spineProtocConfig);
-        AddFields config = spineProtocConfig.getAddFields();
-
         ImmutableList.Builder<CodeGenerationTask> tasks = ImmutableList.builder();
-        EntityStateConfig entityStateConfig = config.getEntityStateConfig();
-        if (isNotDefault(entityStateConfig)) {
-            tasks.add(new GenerateEntityStateFields(entityStateConfig, factory));
+
+        if (spineProtocConfig.hasEntities()) {
```

Comment:
I would consider extracting these `if`s into methods right away, since I'd expect their number grows in the nearest future.